src/compressed_tensors/transforms/base.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Optional, Union
+
+import torch
+from compressed_tensors.registry.registry import RegistryMixin
+
+
+__all__ = ["Transforms"]
+
+
+# TODO: We don't need to save all the __call__ args for serialization or even have
+# them defined by a recipe. Some of them, such as if the transformation should be the
+# first or second matirx in torch.matmul depending on dimensions, can be inferred
+# by the layer time likely.
+
+
+class Transforms(RegistryMixin):
+    def __new__(
+        cls,
+        transform: torch.Tensor,
+        device: Optional[Union[str, torch.device]] = "cpu",
+        *args,
+        **kwargs,
+    ):
+        """
+        Base class for setting up transforms. The registry creates transforms
+        as parameters which can be attached to modules.
+
+        import torch
+
+        size = 1024
+        dtype = torch.bfloat16
+        module = torch.nn.Linear(size, size)
+
+        hadamard_transform = Transforms.load_from_registry(
+            "random_hadamard", size=size, dtype=dtype
+        )
+        hadamard_apply = Transforms.fetch_apply("random_hadamard")
```

Comment:
if we're loading/fetching with two invocations, why not just merge into a single one that returns an object of type `Hadamard`? 
```python
hadamard = Transforms.load_from_registry(
            "random_hadamard", size=size, dtype=dtype
        )
#hadamard_transform corresponds to hadamard.transform
#hadamard_apply corresponds to hadamard.apply
```